### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by rubydevi**
## Description
In this project, I finished creating the remaining classes for school library and created the associations between them.

### Changes Made

- Created a class `Classroom ` with the following:
  - `@label` instance variable, initialized in the constructor.
  - Setter and getter for `@label`.
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following are implemented:
  - Created the `has-many` side.
  - Created the `belongs-to` side.
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables, initialized in the constructor.
  - Setters and getters for instance variables.
- Created a class `Rental` with the following:
  - `@date` instance variable, initialized in the constructor.
  - Setter and getter for `@date`.
- Created the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`. The following has been implemented:
  - Created the `has-many` side of `Book` and `Rental`.
  - Created the `belongs-to` side of `Rental` and `Book`.
  - Created the `has-many` side of `Person` and `Rental`.
  - Created the `belongs-to` side of `Rental` and `Person`.
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

_Thanks for the review in advance_ 🙏

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 
 class Student < Person
-  def initialize(classroom, age, name: 'Unknown', parents_permission: true)
-    super(age, name: name, parents_permission: parents_permission)
+  attr_accessor :classroom
+
+  def initialize(classroom, age, name = 'Unknown', parents_permission: true)
+    super(age, name, parents_permission: parents_permission)
     @classroom = classroom
+    classroom&.add_student(self)
```
Comment (Ibizugbe) [inline student.rb]: Nice job so far👍🏼. Kindly note that it is a requirement that it is a requirement that a method that serves as the 'belongs-to' side (a student belongs to a classroom) should be implemented in the `student` class😉 Kindly do well to implement this👍🏼.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @rubydevi 👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] `a method that serves as the 'has-many' side (a classroom has many students)` has been added👍🏼.


## Required Changes ♻️
A better way to implement the add_rental method in your book class will be to ensure it takes person and date as argument and add itself when adding a new rental see the suggestion below😉.

```suggestion
  def add_rental(date, person)
    Rental.new(date, self, person)
  end
```
Kindly do well to implement that👍🏼

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (peterrobert) [reply student.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi  @rubydevi, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work implementing the belong-to method as requested by the previous reviewer✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rubydevi** [comment]
Thanks for the review 😃, I'll go ahead and merge the PR ☺️

**Pull request merged**